invenio_records_resources/records/systemfields/pid.py

```diff
@@ -106,33 +106,42 @@ def __init__(self, key='id', provider=None, pid_type=None,
             context_cls=context_cls,
         )
 
-    #
-    # Life-cycle hooks
-    #
-    def post_create(self, record):
-        """Called after a record is created."""
-        if self._provider is None or not self._create:
-            return
-
-        # This uses the data descriptor method __get__() below:
-        if getattr(record, self.attr_name) is None:
+    def create(self, record):
+        """Method to create a new persistent identifier for the record."""
+        # This uses the fields __get__() data descriptor method below
+        pid = getattr(record, self.attr_name)
+        if pid is None:
```

Comment:
Minor: It assumes `attr_name` always exists? I think is a fair assumtion since it is assigned in the `SystemField` constructor. However, if not, it will throw an exception...

```
>>> class Test:
...     exists = "yes"
...
>>> getattr(Test, "exists")
'yes'
>>> getattr(Test, "nope")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'Test' has no attribute 'nope'
```

this means that it will fail if it does not exists, it should be: `pid = getattr(record, self.attr_name, None)`, then is harmonized to return None

Reply:
I would leave it as-is, because it's really a big mistake if ``attr_name`` is not set. The property is set automatically when the system field (i.e. data descriptor) is attached to a class via ``__set_name__()``, and not having it means that things should break and not work.